credentials/src/main/kotlin/SSI.kt

```diff
@@ -359,21 +374,86 @@ private fun selectFrom(presentationDefinition: PresentationDefinitionV2, vcJwts:
         }
       }
     }
+
+    return selectableCredentials
   }
 
-  return selectableCredentials
-}
+  /**
+   * Validates a list of Verifiable Credentials (VcJwts) against a presentation definition and generates a presentation submission
+   *
+   * This method checks the presentation definition and the list of Verifiable Credentials to ensure
+   * that the required fields in the definition are satisfied by the provided credentials.
+   *
+   * Currently, if the presentation definition contains submission requirements, or if any required field in an input
+   * descriptor is not satisfied by the provided credentials, an exception is thrown.
+   *
+   * A compliant presentation submission is returned
+   *
+   * ### Throws
+   * - [NotImplementedError] if the Presentation Definition contains Submission Requirements or if a Field Filter is implemented.
+   * - [Exception] if any required field is not satisfied in the given InputDescriptor.
+   *
+   * @param presentationDefinition The [PresentationDefinitionV2] object representing the presentation's definition.
+   * @param vcJwts The list of [VcJwt] representing the Verifiable Credentials to be validated against the presentation definition.
+   * @throws Exception If a required field is not satisfied in a given InputDescriptor.
+   * @return the generated presentation submission.
+   */
+  @Throws(Exception::class)
+  private fun generatePresentationSubmissionFrom(
+    presentationDefinition: PresentationDefinitionV2,
+    vcJwts: List<VcJwt>
+  ): PresentationSubmission {
+    if (!presentationDefinition.submissionRequirements.isNullOrEmpty()) {
+      throw NotImplementedError("Presentation Definition's Submission Requirements feature is not implemented")
+    }
 
-private fun evaluateJsonPath(vcJwt: VcJwt, path: String): String? {
-  val vc: VerifiableCredentialType =
-    FromJwtConverter.fromJwtVerifiableCredential(JwtVerifiableCredential.fromCompactSerialization(vcJwt))
+    val descriptorMapList = mutableListOf<DescriptorMap>()
 
-  val vcJsonString: String = vc.toJson()
-  val result: String? = JsonPath.parse(vcJsonString)?.read<String>(path)
+    for (inputDescriptor: InputDescriptorV2 in presentationDefinition.inputDescriptors) {
+      if (inputDescriptor.constraints.fields!!.isNotEmpty()) {
+        for (field: FieldV2 in inputDescriptor.constraints.fields) {
+          if (field.optional == true) continue // Skip optional fields
 
-  return result
+          var fieldMatch = false
+          for ((vcIndex, vcJwt) in vcJwts.withIndex()) {
+            for (path: String in field.path) {
+              val jsonPathResult = evaluateJsonPath(vcJwt, path)
+              if (jsonPathResult != null) {
```

Comment:
I think this might simplify the code
```suggestion
            for (jsonPathResult: String in field.path.mapNotNull{ evaluateJsonPath(vcJwt, it) }) {
```